### DISCUSSION TITLE
각 class 별로 AP 확인하기

### DISCUSSION THREAD

**Discussion by jaehwan-AI** [boostcampaitech2/object-detection-level2-cv-06#125] [ideas]
학습의 결과로 mAP는 확인할 수 있었지만 과연 내 모델이 어떤 class를 잘 맞추고 있는지 궁금하지 않으셨나요?
간단한 코드 수정으로 wandb랑 연동까지는 아니더라도 학습 log에 각 class 별로 AP가 찍히게는 할 수 있었습니다.

mmdetection/mmdet/datasets/coco.py 에 가시면 CocoDataset 클래스에 evaluate 라는 함수가 있습니다.
![image](https://user-images.githubusercontent.com/66052001/136777366-ee880eaf-5d0d-4bd8-9777-f67f1910b51a.png)

함수의 **classwise를 True**로 바꿔주시면 학습 시 class 별로 AP를 다음과 같이 log에 찍어줍니다.
![image](https://user-images.githubusercontent.com/66052001/136777421-8c3eba44-fe8f-48e4-8672-ef24efaaff83.png)

혹시 다른 방안이 있다면 댓글로 알려주시면 감사하겠습니다!

**[+추가사항]**
![image](https://user-images.githubusercontent.com/66052001/136871122-96c0885c-934a-46e2-a2a7-4f50fc30b0c0.png)
이렇게도 사용이 가능하다고 합니다

**Comment by kmouleejunhyuk** [comment]
배터리보다 trash AP가 더 적은게 신기하네요. 너무 다양해서 그럴까요..

**Reply by jaehwan-AI** [reply]
+---------------+-------+-------------+-------+------------+-------+
| category      | AP    | category    | AP    | category   | AP    |
+---------------+-------+-------------+-------+------------+-------+
| General trash | 0.159 | Paper         | 0.294 | Paper pack | 0.311 |
| Metal             | 0.389 | Glass          | 0.342 | Plastic        | 0.256 |
| Styrofoam      | 0.348 | Plastic bag | 0.537 | Battery      | 0.575 |
| Clothing         | 0.427 | None         | None  | None        | None  |
+---------------+-------+-------------+-------+------------+-------+
위 결과가 faster r-cnn 24에폭 학습한 결과인데 확실히 수가 많고 다양한 General trash랑 Paper, Plastic이 AP가 적어요..
잘못 라벨링 된 데이터가 많은 class일 가능성도 있고, 여러 bbox가 겹치는 라벨링이 많은 class일 수도 있을 것 같아요

**Reply by JiyouSeo** [reply]
오...엄청 꿀팁이네요 공유 감사드립니다!!

**Reply by kwakjeeyoon** [reply]
오오!! 공유해주셔서 감사합니다!!
